codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.http2;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoop;
+import io.netty.channel.EventLoopGroup;
+import io.netty.util.AttributeKey;
+import io.netty.util.internal.UnstableApi;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * A class that makes it easy to bootstrap a new HTTP/2 stream as a {@link Channel}.
+ *
+ * <p>The bootstrap requires a registered parent {@link Channel} with a {@link ChannelPipeline} that contains the
+ * {@link Http2MultiplexCodec}.
+ *
+ * <p>A child channel becomes active as soon as it is registered to an eventloop. Therefore, an active channel does not
+ * map to an active HTTP/2 stream immediately. Only once a {@link Http2HeadersFrame} has been sent or received, does
+ * the channel map to an active HTTP/2 stream. In case it was not possible to open a new HTTP/2 stream (i.e. due to
+ * the maximum number of active streams being exceeded), the child channel receives an exception indicating the reason
+ * and is closed immediately thereafter.
+ *
+ * <p>This class is thread-safe.
+ */
+// TODO(buchgr): Should we deliver a user event when the stream becomes active? For all stream states?
```

Comment:
using the `ChannelHandler handler` member variable in this class allows the user to know when the child channel is created, and the user can use the close future on the channel to determine when it is closed ... maybe this is good enough for now. I don't have a good feeling yet on whether it is necessary / useful to notify for all state transitions but we can add this later when necessary.